### PR TITLE
CRI-O, Docker: set the tag to the OpenShift release on RHEL

### PR DESCRIPTION
**Pull request by giuseppe**
Closes: https://bugzilla.redhat.com/show_bug.cgi?id=1493376

### CODE REVIEW

**Comment by giuseppe** [comment]
Blocked on: https://github.com/openshift/openshift-ansible/pull/5354

File: roles/docker/tasks/systemcontainer_crio.yml [hunk 1]
```diff
@@ -104,8 +105,9 @@
 
     - name: Use RHEL based image when distribution is Red Hat
       set_fact:
-        l_crio_image_prepend: "registry.access.redhat.com"
+        l_crio_image_prepend: "registry.access.redhat.com/openshift3"
         l_crio_image_name: "cri-o"
+        l_docker_image_tag: "{{ openshift_image_tag }}"
       when: ansible_distribution == "RedHat"
```
Comment (sdodson) [inline roles/docker/tasks/systemcontainer_crio.yml]: Anyone running RHEL and installing origin would be pulling CRI-O from the openshift3 namespace, I believe we should check for `openshift_deployment_type == "openshift-enterprise"` instead.

File: roles/docker/tasks/systemcontainer_docker.yml [hunk 2]
```diff
@@ -89,10 +89,12 @@
     - name: Set to default prepend
       set_fact:
         l_docker_image_prepend: "gscrivano"
+        l_docker_image_tag: "latest"
 
     - name: Use Red Hat Registry for image when distribution is Red Hat
       set_fact:
         l_docker_image_prepend: "registry.access.redhat.com/openshift3"
+        l_docker_image_tag: "{{ openshift_image_tag }}"
       when: ansible_distribution == 'RedHat'
```
Comment (sdodson) [inline roles/docker/tasks/systemcontainer_docker.yml]: Same comment as above.

**Comment by giuseppe** [comment]
@sdodson there is a circular dependency if I'd use the `openshift_version` role (as it uses docker)  to handle correctly `openshift_version`.  Do you have any suggestion on how this could be done?

**Comment by giuseppe** [comment]
I've splitted this PR and moved the safe changes here: https://github.com/openshift/openshift-ansible/pull/5519

**Comment by ashcrow** [comment]
#5354 merged.

@sdodson PTAL at @giuseppe's question.

**Comment by ashcrow** [comment]
Totally unrelated:
```
        "Error from server (Forbidden): User \"system:admin\" cannot create imagestreams in the namespace \"openshift\": User \"system:admin\" cannot create imagestreams in project \"openshift\": clusterrole.rbac.authorization.k8s.io \"cluster-admin\" not found (post imagestreams)", 
```

**Comment by sdodson** [comment]
> @sdodson there is a circular dependency if I'd use the openshift_version role (as it uses docker) to handle correctly openshift_version. Do you have any suggestion on how this could be done?

I think we should start requiring that openshift_release is set and set the image tag to that value prefixed with v if necessary. We should put a card in to make openshift_release required and to strip out all the insane things we're doing in order to determine release version automatically.

**Comment by ashcrow** [comment]
@sdodson @giuseppe Card created https://trello.com/c/Z83qZeDi/530-require-openshiftrelease

**Comment by ashcrow** [comment]
@sdodson so would that work then block this work? Or is there a temporary solution that you'd recommend between now and then?

**Comment by sdodson** [comment]
I'd say start relying on it and add it to the sanitize inventory bits to ensure that it's required as part of this PR. Then we can do all the followup cleanup within the context of the card you opened.

**Comment by giuseppe** [comment]
I've added a check for `openshift_image_tag`.  It cannot be merged yet, I don't see the v3.7 tags for the crio and the container-engine system containers.
/cc @jwhonce

**Comment by ashcrow** [comment]
Containerized fail is a flake AFAICT. Tox looks to be legit:

```
roles/docker/tasks/main.yml:
  17:5      error    wrong indentation: expected 6 but found 4  (indentation)
  24:3      error    wrong indentation: expected 4 but found 2  (indentation)
  30:3      error    wrong indentation: expected 4 but found 2  (indentation)
  36:3      error    wrong indentation: expected 4 but found 2  (indentation)

roles/docker/tasks/systemcontainer_crio.yml:
  111:7     error    wrong indentation: expected 8 but found 6  (indentation)

roles/docker/tasks/systemcontainer_docker.yml:
  99:7      error    wrong indentation: expected 8 but found 6  (indentation)
```

**Comment by giuseppe** [comment]
I fixed the tox issues.  I think it is ready once we have the tags on the CRI-O image

**Comment by giuseppe** [comment]
the CRI-O image seems to have the correct tags now, I think this PR is ready to be merged

**Pull request merged**